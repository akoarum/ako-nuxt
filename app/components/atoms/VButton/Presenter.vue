<script>
export default {
  props: {
    label: { type: String, required: true },
    primary: { type: Boolean },
    secondary: { type: Boolean },
    small: { type: Boolean }
  },
  computed: {
    appearanceClasses() {
      const classes = []

      if (this.primary) {
        classes.push('-primary')
      }

      if (this.secondary) {
        classes.push('-secondary')
      }

      if (this.small) {
        classes.push('-small')
      }

      return classes
    }
  }
}
</script>
