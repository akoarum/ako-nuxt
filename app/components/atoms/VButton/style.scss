.btn {
  background: $color-text;
  line-height: 1;
  color: #fff;
  transition: background $transition-duration $transition-easing;
  will-change: background;

  &.-primary {
    background: $color-primary;
  }
  &.-secondary {
    background: $color-secondary;
  }
  &:disabled {
    background: $color-disabled;
  }

  @media screen and (min-width: 768px) {
    border-radius: 5px;
    padding: 8px 24px 9px;
    font-size: $font-size-base;

    &.-small {
      padding: 6px 20px;
      font-size: 1.4rem;
    }

    &:hover,
    &:focus {
      background: #676767;

      &.-primary {
        background: #47a4f3;
      }
      &.-secondary {
        background: #ffa450;
      }
      &:disabled {
        background: $color-disabled;
      }
    }
  }
  @media screen and (max-width: 767px) {
    box-sizing: border-box;
    border-radius: 6px;
    padding: 11px 24px 12px;
    width: 100%;
    font-size: $font-size-base;

    &.-small {
      padding: 8px 24px;
    }

    &:hover,
    &:focus {
      background: #676767;

      &.-primary {
        background: #47a4f3;
      }
      &.-secondary {
        background: #ffa450;
      }
      &:disabled {
        background: $color-disabled;
      }
    }
  }
}
