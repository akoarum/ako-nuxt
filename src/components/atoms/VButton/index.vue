<template>
  <button
    v-bind="$attrs"
    :class="styledClass"
    type="button"
    class="btn"
    @click="$emit('click')"
  >
    <slot />
  </button>
</template>

<script>
export default {
  name: 'VButton',
  inheritAttrs: true,
  props: {
    secondary: { type: Boolean },
    caution: { type: Boolean }
  },
  computed: {
    styledClass() {
      if (this.secondary) return '-secondary'
      if (this.caution) return '-caution'
      return ''
    }
  }
}
</script>

<style lang="scss" scoped>
@import './style';
</style>
