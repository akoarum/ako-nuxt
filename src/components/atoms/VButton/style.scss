.btn {
  display: inline-block;
  border-radius: 2px;
  box-sizing: border-box;
  width: 100%;
  background: $color-theme;
  font-size: $fontSize-pc;
  line-height: 1.3;
  color: #fff;
  text-align: center;
  transition: background 0.3s $easing;

  @include media-min {
    padding: 9px 14px 8px;
    max-width: 160px;

    &:hover,
    &:focus {
      background: #5dc5e8;
    }
  }

  @include media-max {
    padding: 10px 16px 9px;
  }
}

.-secondary {
  background: $color-secondary;

  @include media-min {
    &:hover,
    &:focus {
      background: #49bb99;
    }
  }
}

.-caution {
  background: $color-caution;

  @include media-min {
    &:hover,
    &:focus {
      background: #f75656;
    }
  }
}

.btn:disabled {
  background: $color-notice;
  cursor: default;

  &:hover,
  &:focus {
    background: $color-notice;
  }
}
