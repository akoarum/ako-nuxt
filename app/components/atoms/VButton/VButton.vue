<template>
  <button class="btn" :class="appearanceClasses" @click="$emit('click')">
    <span>{{ label }}</span>
  </button>
</template>

<script>
import Presenter from './Presenter'

export default {
  mixins: [Presenter]
}
</script>

<style lang="scss" scoped>
@import './style';
</style>
