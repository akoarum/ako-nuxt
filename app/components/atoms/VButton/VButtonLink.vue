<template>
  <router-link :to="to" class="btn" :class="appearanceClasses">
    <span>{{ label }}</span>
  </router-link>
</template>

<script>
import Presenter from './Presenter'
export default {
  mixins: [Presenter],
  props: {
    to: { type: [String, Object], required: true }
  }
}
</script>

<style lang="scss" scoped>
@import './style';
</style>
