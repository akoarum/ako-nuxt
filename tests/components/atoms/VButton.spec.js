import { shallowMount } from '@vue/test-utils'
import VButton from '~/components/atoms/VButton'

describe('VButton', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(VButton, {
      propsData: {}
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('computed', () => {
    it('styledClass: propsからクラスを生成して返す', () => {
      expect(VButton.computed.styledClass.call({ secondary: true })).toBe('-secondary')
      expect(VButton.computed.styledClass.call({ caution: true })).toBe('-caution')
    })
  })
})
