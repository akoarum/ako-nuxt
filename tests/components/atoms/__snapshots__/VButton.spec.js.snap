// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VButton initialized snapshot 1`] = `
<button
  class="btn"
  type="button"
/>
`;
